tests/integration/test_charm.py

```diff
@@ -96,6 +97,33 @@ async def test_attach_services(ops_test: OpsTest):
     assert unit.workload_status == BlockedStatus.name
 
 
+async def test_detach_before_attach(ops_test: OpsTest):
+    charm = ops_test.model.applications[CHARM_NAME]
+    unit = charm.units[0]
+
+    # Attach to pro subscription with services "manually" to get the unit
+    # in a weird state to test the detach before attach logic.
+    action = await unit.run(f"sudo pro attach {TEST_TOKEN} --no-auto-enable")
+    await action.wait()
+    action = await unit.run("sudo pro api u.pro.status.is_attached.v1")
+    await action.wait()
+    assert json.loads(action.results["stdout"])["data"]["attributes"]["is_attached"] is True
+
+    # Attach to pro subscription with services
+    # This will detach the existing attachment and then attach.
+    # We will know the second attachment succeeded because it will enable esm-infra.
+    await charm.set_config({"services": "esm-infra", "token": TEST_TOKEN})
```

Comment:
Should we also perform a check with the `enabled_services` endpoint to be extra sure that `esm-infra` was indeed enabled here